### ISSUE TITLE
The next steps of development: RC1 > RC2 > RC3 > Final Release

### ISSUE THREAD

**Issue by davideas** [davideas/FlexibleAdapter#361]
A new clearer plan for the next releases. Please follow the wiki page [Migrations](wiki/Migrations) for all braking changes, between those releases.

Here, the main points of **rc1** milestone:
- [x] I'm working on the new big feature of **Scrollable Headers** and **Footers**. Since this feature brings lot of new code, it must be closed as soon as possible.
- [x] I need to improve the **EndlessScrolling / LoadingMore**: how it is now, it is not good and with the new concept of footers the _progressItem_ could become a footer item.
- [x] I need to fix 2 small but annoying bugs in **scrolling animations**.
- [x] I have to **refactor / deprecate** many functions, however, they will be removed in the rc3/final release. So this also must be done asap too.
- [x] I can simplify a lot the **configuration** for the **sticky header container**, doing so we can remove the mini layout from ours xmls.
- [x] Need to add the possibility to set **elevation** to the _header item_ and to the _sticky header container_.

For **rc2** milestone:
- [x] Simplify the initialization and review internal calculations for updateDataSet and filtering.
- [x] I need to evaluate the creation of a separate library for the extensions.
- [x] Extension to support **DataBinding** as this seems highly demanded by the users.
- [x] Support of **Endless Top Scrolling**
- [x] Support of third LayoutManagers.
- [x] List copy managed internally.
- [x] An huge improvement for the **FastScroller**.
- [x] Get rid of _Method B_.
- [ ] Evaluating the possibility to filter items saving internally the originalList.
- [ ] Code optimization.
- [ ] A new and simple to read demoApp (anticipated from final release).
- [ ] More example for the demoApp.
- [ ] More unit tests.
- [ ] Finish the **Wiki pages**.
- [x] **Bug fixes** that come from rc1 pre-release.

The **rc3** milestone has big challenges:
In order to support the DB extensions a new level of Abstraction should be designed. Those 3 extensions will not be confirmed once an acceptable class has been produced.
- [ ] Extension to support **Realm** (moved from rc2).
- [ ] Extension to support **Firebase** (moved from rc2).
- [ ] Extension to support **Cursor** (as it seems highly demanded by the users).
- [ ] Extension for **UI**: some classes might be moved into this library and new one will come (see next point).
- [ ] I will see what I can do to improve swipe capability.
- [ ] Dialogs with Adapter and RecyclerView.
- [ ] StickyHeaders on the left side.
- [ ] A light Adapter with basic functionalities.
- [ ] Delete of all deprecated functions (moved from final, we are getting close to final release!).
- [ ] **Bug fixes** that come from rc2 pre-release.

For **final** release:
- [ ] Some other code adjustment that will appear clearer with time.
- [ ] Some bug fixes that come from rc3 pre-release.
- [ ] Full review of the demoApp and part of the documentation.
- [ ] Publishing to PlayStore.

**Comment by livotov** [comment]
It looks like RC3 falls into a separate 1.1 release - too many architect work to be done. Shouldn't you better release 1.0, then focus on described extensions for 1.1 instead (as I suppose they could take a long time for a proper, stable and well done implementation) ?

Otherwise - a great work ! :)
Thanks!!!

**Comment by tprochazka** [comment]
I'm personally very like for such number of new great features!
Just maybe one RC should be split to the several full releases ;-)
RC is Release Candidate, there should be just necessary critical bug fixes not completely new breaking features ;-)

Now I'm using SNAPSHOT on my project and it is not the ideal way, it can break my builds every day and also tagging commits of my app like the release is very problematic.

**Comment by djuelg** [comment]
I agree that it would be more convenient for your library users to publish a new usable version and publish the other features/refactorings as a seperated release.

**Comment by davideas** [comment]
Dears all, OK, I'm convinced from you all to review the plan for the releases. Indeed, the extensions will be new libraries with their own versioning. So, it's better to finalize and make stable this development as you proposed.
I will publish the RC2 as soon as possible.
I will keep only few point and move the development of new demoApp in parallel with RC3.

**Comment by davideas** [comment]
**News for coming RC3**

- I do not have time to rewrite the demoApp for the RC3 pre-release.
- Regarding the new swipe capability, I still don't find a good alternative: the touch listener seems to be the solution most used by others libraries, but here it will require a deep change and lot of time that now I can't dedicate.

So we will for the RC3 soon.

**Comment by davideas** [comment]
I will produce a **RC4** as extra release due to important bug fixed.

**Comment by davideas** [comment]
Final version is foreseen on 04 March 2018.
Unfortunately, the new demoApp will follow according to my free time.
